plugin/storage/es/spanstore/reader.go

```diff
@@ -238,19 +238,40 @@ func (s *SpanReader) FindTraces(traceQuery *spanstore.TraceQueryParameters) ([]*
 	if err != nil {
 		return nil, err
 	}
-	var retMe []*model.Trace
-	for _, traceID := range uniqueTraceIDs {
-		if len(retMe) >= traceQuery.NumTraces {
-			break
-		}
-		trace, err := s.readTrace(traceID, traceQuery)
+	return s.multiRead(uniqueTraceIDs, traceQuery)
+}
+
+func (s *SpanReader) multiRead(traceIDs []string, traceQuery *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
+	if len(traceIDs) == 0 {
+		return []*model.Trace{}, nil
+	}
+	searchRequests := make([]*elastic.SearchRequest, len(traceIDs))
+
+	for i := range searchRequests {
```

Comment:
I prefer: `for i, traceID := range traceIDs` so you don't have to do index on L251